src/test/java/com/google/step/youtube/DescriptionParserTest.java

```diff
@@ -307,148 +339,156 @@ public void codeWithQuotes_multipleMatches() {
-        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
-        assertThat(actual, equalTo(Arrays.asList("https://lmg.gg/glasswire")));
+        List<OfferSnippet> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(extractPromoCodes(actual), equalTo(Arrays.asList("https://lmg.gg/glasswire")));
     }
 
     @Test
     public void toAtLinks_stopMatchingAtSpace() {
         String desc = "Head to https://www.squarespace.com/boulderin... to save 10% off your "
                 + "first purchase of a website or domain using code BOULDERINGBOBAT. ";
-        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
-        assertThat(actual, equalTo(Arrays.asList("https://www.squarespace.com/boulderin...")));
+        List<OfferSnippet> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(extractPromoCodes(actual), equalTo(Arrays.asList("https://www.squarespace.com/boulderin...")));
     }
 
     @Test
     public void toAtLinks_stopMatchingAtNewline() {
         String desc = "Head to https://www.squarespace.com/boulderin...\nto save 10% off your "
                 + "first purchase of a website or domain using code BOULDERINGBOBAT. ";
-        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
-        assertThat(actual, equalTo(Arrays.asList("https://www.squarespace.com/boulderin...")));
+        List<OfferSnippet> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(extractPromoCodes(actual), equalTo(Arrays.asList("https://www.squarespace.com/boulderin...")));
     }
 
     @Test
     public void toAtLinks_stopMatchingAtComma() {
         String desc = "Go to http://stamps.com, click on the microphone";
-        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
-        assertThat(actual, equalTo(Arrays.asList("http://stamps.com")));
+        List<OfferSnippet> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(extractPromoCodes(actual), equalTo(Arrays.asList("http://stamps.com")));
     }
 
     @Test
     public void toAtLinks_stopMatchingAtCloseParens() {
         String desc = "Buy or sell almost anything on Mercari on the App store or at http://mercari.com)";
-        List<String> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
-        assertThat(actual, equalTo(Arrays.asList("http://mercari.com")));
+        List<OfferSnippet> actual = DescriptionParser.findMatches(TO_AT_LINKS_PATTERN, desc);
+        assertThat(extractPromoCodes(actual), equalTo(Arrays.asList("http://mercari.com")));
     }
 
+
+    // helper method for findMatches() tests
+    private List<String> extractPromoCodes(List<OfferSnippet> offerSnippets) {
+        List<String> promoCodes = new ArrayList<>();
+        offerSnippets.forEach( (snippet) -> promoCodes.add(snippet.getPromoCode()));
```

Comment:
i think you want:
```
offerSnippets.stream().map(OfferSnippet::getPromoCode).collect(ImmutableList.toImmutableList())
```

also i guess you're not checking the other fields in the value type?

Reply:
Changed and added a comment about only checking the promocode field.